webapp/src/pages/Settings.tsx

```diff
@@ -557,133 +637,292 @@ const Settings: React.FC<Props> = ({ open, onClose }) => {
+                                ) ? (
+                                  <>
+                                    <StringField
+                                      label="remote"
+                                      nullable
+                                      value={postprocessor.remote}
+                                      disabled={isUpdatingConfig}
+                                      onChange={(remote) =>
+                                        updatePostprocessor(
+                                          pipelineIndex,
+                                          index,
+                                          {
+                                            remote,
+                                          }
+                                        )
+                                      }
+                                    />
+                                    <JSONField
+                                      array
+                                      label="args"
+                                      value={postprocessor.args}
+                                      disabled={isUpdatingConfig}
+                                      onChange={(args) =>
+                                        updatePostprocessor(
+                                          pipelineIndex,
+                                          index,
+                                          {
+                                            args,
+                                          }
+                                        )
+                                      }
+                                    />
+                                    <JSONField
+                                      label="kwargs"
+                                      value={postprocessor.kwargs}
+                                      disabled={isUpdatingConfig}
+                                      onChange={(kwargs) =>
+                                        updatePostprocessor(
+                                          pipelineIndex,
+                                          index,
+                                          {
+                                            kwargs,
+                                          }
+                                        )
+                                      }
+                                    />
+                                  </>
+                                ) : (
+                                  (postprocessor.class_name?.includes(
+                                    "Temperature"
+                                  ) && (
+                                    <NumberField
+                                      label="temperature"
+                                      value={
+                                        "temperature" in postprocessor
+                                          ? postprocessor.temperature
+                                          : 0.5
```

Comment:
3 things:
1. That should be equivalent to
```suggestion
                                        postprocessor.temperature ?? 0.5
```
2. I think 1 would be a better default for temperature, as it is equivalent to not applying any temperature scaling. Starting from there, the user would choose if it should go up or down.
3. If you change from a "known" postprocessor to another, the `temperature` and `threshold` don't get cleaned up, so you end up with an invalid config that fails:
```json
{
    "class_name": "azimuth.utils.ml.postprocessing.TemperatureScaling",
    "args": [],
    "kwargs": {
        "temperature": 0.6
    },
    "remote": null,
    "threshold": 0.5,
    "temperature": 0.6
}
```

Reply:
Your solution doesn't seem to work. I am not able to define a custom `class_name` anymore (with `args`, `kwargs`, etc.).

Reply:
Can you give please some examples to define the class name with a custom value? I can figure out the issue behind it.

Reply:
I don't have a concrete example, but if I type "potato" as the `class_name`, I expect to get the `args`, `kwargs` and `remote` fields.

Reply:
Oh! I see that it works, I just have to hit enter on the `class_name` field. That was not obvious to me. Let's think of a solution...

Reply:
Okay, Yes! Thank you for confirming it. I am figuring out to skip enter key.

Reply:
I found that by adding `autoSelect`, we don't need to hit enter, just to focus out of the field. 👌 
With that, I found that setting `isOptionEqualToValue={() => false}` helps make sure there isn't an option that is highlighted by accident (which would get unexpectedly selected when focusing out).
Also, I would add `disableClearable` to remove the `x` button as it doesn't seem necessary.

Reply:
Thank you for the suggestion :)